src/DropDown/DropDown.component.tsx

```diff
@@ -60,48 +81,122 @@ const StyledDropDownContainer = styled(DropDownContainer)<
-    padding: 0.25rem 0;
+    padding: 0.25rem auto;
     z-index: 1;
     background-color: ${th.color('neutrals.white.base')};
-    box-shadow: 0 0.5rem 0.5rem 0 ${th.color('neutrals.ash.base')};
+    box-shadow: 0 0 0.5rem 0 ${th.color('neutrals.ash.base')};
     border-radius: base;
     border: thin solid ${th.color('neutrals.silver.dark')};
+
     /* Defined Position */
-    top: ${({ position: { top } = {} }) =>
-        top ? `${top}px !important` : 'inherit'};
-    left: ${({ position: { left } = {} }) =>
-        left ? `${left}px !important` : '0'};
-    right: ${({ position: { right } = {} }) =>
-        right ? `${right}px !important` : 'inherit'};
-    bottom: ${({ position: { bottom } = {} }) =>
-        bottom ? `${bottom}px !important` : 'inherit'};
+    ${({ position, height, width, offset = '0.5rem' }) => {
+        const CSS: any = {};
+        switch (position) {
+            case 'bottom':
+                CSS.top = `${height}px`;
+                CSS.left = 0;
+                break;
+            case 'right':
+                CSS.left = `${width}px`;
+                CSS.top = '-0.75rem';
+                break;
+            case 'left':
+                CSS.right = `${width}px`;
+                CSS.top = '-0.75rem';
+                break;
+            case 'top':
+                CSS.bottom = `${height}px`;
+                CSS.left = 0;
+                break;
+            default:
+                break;
+        }
+        return css(CSS);
+    }};
 `;
 
-const ArrowUp = styled('div')`
+const ArrowBase = css`
     position: absolute;
-    top: -0.4rem;
-    left: 1rem;
     width: 0;
     height: 0;
-    border-left: 0.75rem solid transparent;
-    border-right: 0.75rem solid transparent;
-    border-bottom: 0.5rem solid white;
 `;
 
+const CaretHeight = '0.4rem';
+const CaretIndent = '0.75rem';
+const CaretSize = '0.5rem';
+
+const ArrowUp = styled('div')`
+    ${ArrowBase};
+    top: -${CaretHeight};
+    left: ${CaretIndent};
+    border-left: ${CaretSize} solid transparent;
+    border-right: ${CaretSize} solid transparent;
+    border-bottom: ${CaretSize} solid white;
+`;
+
+const ArrowDown = styled('div')`
+    ${ArrowBase};
+    bottom: -${CaretHeight};
+    left: ${CaretIndent};
+    border-left: ${CaretSize} solid transparent;
+    border-right: ${CaretSize} solid transparent;
+    border-top: ${CaretSize} solid white;
+`;
+const ArrowRight = styled('div')`
+    ${ArrowBase};
+    top: ${CaretIndent};
+    right: -${CaretHeight};
+    border-top: ${CaretSize} solid transparent;
+    border-bottom: ${CaretSize} solid transparent;
+    border-left: ${CaretSize} solid white;
+`;
+
+const ArrowLeft = styled('div')`
+    ${ArrowBase};
+    top: ${CaretIndent};
+    left: -${CaretHeight};
+    border-top: ${CaretSize} solid transparent;
+    border-bottom: ${CaretSize} solid transparent;
+    border-right: ${CaretSize} solid white;
+`;
+
+const renderArrow = (position: Position, isShown?: boolean) => {
```

Comment:
There is a lot of logic/syntax just for the arrow, plus here its parts are assembled into a functional component. Perhaps move to its own file?